commit 3de2f8755c6c036dcd0b1f4acd6bcdefe0e775c0
Author: Joseph Myers <josmyers@redhat.com>
Date:   Wed Mar 13 13:57:56 2024 +0000

    Update syscall lists for Linux 6.8
    
    Linux 6.8 adds five new syscalls.  Update syscall-names.list and
    regenerate the arch-syscall.h headers with build-many-glibcs.py
    update-syscalls.
    
    Tested with build-many-glibcs.py.

    Modified for RHEL by: Patsy Griffin <patsy@redhat.com>

diff --git a/sysdeps/unix/sysv/linux/aarch64/arch-syscall.h b/sysdeps/unix/sysv/linux/aarch64/arch-syscall.h
index 1713897f85..7ee8a2167a 100644
--- a/sysdeps/unix/sysv/linux/aarch64/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/aarch64/arch-syscall.h
@@ -122,12 +122,16 @@
 #define __NR_lgetxattr 9
 #define __NR_linkat 37
 #define __NR_listen 201
+#define __NR_listmount 458
 #define __NR_listxattr 11
 #define __NR_llistxattr 12
 #define __NR_lookup_dcookie 18
 #define __NR_lremovexattr 15
 #define __NR_lseek 62
 #define __NR_lsetxattr 6
+#define __NR_lsm_get_self_attr 459
+#define __NR_lsm_list_modules 461
+#define __NR_lsm_set_self_attr 460
 #define __NR_madvise 233
 #define __NR_map_shadow_stack 453
 #define __NR_mbind 235
@@ -276,6 +280,7 @@
 #define __NR_socketpair 199
 #define __NR_splice 76
 #define __NR_statfs 43
+#define __NR_statmount 457
 #define __NR_statx 291
 #define __NR_swapoff 225
 #define __NR_swapon 224
diff --git a/sysdeps/unix/sysv/linux/alpha/arch-syscall.h b/sysdeps/unix/sysv/linux/alpha/arch-syscall.h
index 5457d2d8ae..0f4ea7670b 100644
--- a/sysdeps/unix/sysv/linux/alpha/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/alpha/arch-syscall.h
@@ -150,12 +150,16 @@
 #define __NR_link 9
 #define __NR_linkat 458
 #define __NR_listen 106
+#define __NR_listmount 568
 #define __NR_listxattr 388
 #define __NR_llistxattr 389
 #define __NR_lookup_dcookie 406
 #define __NR_lremovexattr 392
 #define __NR_lseek 19
 #define __NR_lsetxattr 383
+#define __NR_lsm_get_self_attr 569
+#define __NR_lsm_list_modules 571
+#define __NR_lsm_set_self_attr 570
 #define __NR_lstat 68
 #define __NR_lstat64 426
 #define __NR_madvise 75
@@ -441,6 +445,7 @@
 #define __NR_stat64 425
 #define __NR_statfs 328
 #define __NR_statfs64 528
+#define __NR_statmount 567
 #define __NR_statx 522
 #define __NR_swapoff 304
 #define __NR_swapon 322
diff --git a/sysdeps/unix/sysv/linux/arc/arch-syscall.h b/sysdeps/unix/sysv/linux/arc/arch-syscall.h
index a66471c83a..90359482a8 100644
--- a/sysdeps/unix/sysv/linux/arc/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/arc/arch-syscall.h
@@ -126,12 +126,16 @@
 #define __NR_lgetxattr 9
 #define __NR_linkat 37
 #define __NR_listen 201
+#define __NR_listmount 458
 #define __NR_listxattr 11
 #define __NR_llistxattr 12
 #define __NR_llseek 62
 #define __NR_lookup_dcookie 18
 #define __NR_lremovexattr 15
 #define __NR_lsetxattr 6
+#define __NR_lsm_get_self_attr 459
+#define __NR_lsm_list_modules 461
+#define __NR_lsm_set_self_attr 460
 #define __NR_madvise 233
 #define __NR_map_shadow_stack 453
 #define __NR_mbind 235
@@ -278,6 +282,7 @@
 #define __NR_socketpair 199
 #define __NR_splice 76
 #define __NR_statfs64 43
+#define __NR_statmount 457
 #define __NR_statx 291
 #define __NR_swapoff 225
 #define __NR_swapon 224
diff --git a/sysdeps/unix/sysv/linux/arm/arch-syscall.h b/sysdeps/unix/sysv/linux/arm/arch-syscall.h
index 74a57f4520..4930167a03 100644
--- a/sysdeps/unix/sysv/linux/arm/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/arm/arch-syscall.h
@@ -164,12 +164,16 @@
 #define __NR_link 9
 #define __NR_linkat 330
 #define __NR_listen 284
+#define __NR_listmount 458
 #define __NR_listxattr 232
 #define __NR_llistxattr 233
 #define __NR_lookup_dcookie 249
 #define __NR_lremovexattr 236
 #define __NR_lseek 19
 #define __NR_lsetxattr 227
+#define __NR_lsm_get_self_attr 459
+#define __NR_lsm_list_modules 461
+#define __NR_lsm_set_self_attr 460
 #define __NR_lstat 107
 #define __NR_lstat64 196
 #define __NR_madvise 220
@@ -361,6 +365,7 @@
 #define __NR_stat64 195
 #define __NR_statfs 99
 #define __NR_statfs64 266
+#define __NR_statmount 457
 #define __NR_statx 397
 #define __NR_swapoff 115
 #define __NR_swapon 87
diff --git a/sysdeps/unix/sysv/linux/csky/arch-syscall.h b/sysdeps/unix/sysv/linux/csky/arch-syscall.h
index ba7632e018..3f16a29f57 100644
--- a/sysdeps/unix/sysv/linux/csky/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/csky/arch-syscall.h
@@ -131,12 +131,16 @@
 #define __NR_lgetxattr 9
 #define __NR_linkat 37
 #define __NR_listen 201
+#define __NR_listmount 458
 #define __NR_listxattr 11
 #define __NR_llistxattr 12
 #define __NR_llseek 62
 #define __NR_lookup_dcookie 18
 #define __NR_lremovexattr 15
 #define __NR_lsetxattr 6
+#define __NR_lsm_get_self_attr 459
+#define __NR_lsm_list_modules 461
+#define __NR_lsm_set_self_attr 460
 #define __NR_madvise 233
 #define __NR_map_shadow_stack 453
 #define __NR_mbind 235
@@ -291,6 +295,7 @@
 #define __NR_socketpair 199
 #define __NR_splice 76
 #define __NR_statfs64 43
+#define __NR_statmount 457
 #define __NR_statx 291
 #define __NR_swapoff 225
 #define __NR_swapon 224
diff --git a/sysdeps/unix/sysv/linux/hppa/arch-syscall.h b/sysdeps/unix/sysv/linux/hppa/arch-syscall.h
index 483706de9b..a1b2c819d6 100644
--- a/sysdeps/unix/sysv/linux/hppa/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/hppa/arch-syscall.h
@@ -155,12 +155,16 @@
 #define __NR_link 9
 #define __NR_linkat 283
 #define __NR_listen 32
+#define __NR_listmount 458
 #define __NR_listxattr 244
 #define __NR_llistxattr 245
 #define __NR_lookup_dcookie 223
 #define __NR_lremovexattr 248
 #define __NR_lseek 19
 #define __NR_lsetxattr 239
+#define __NR_lsm_get_self_attr 459
+#define __NR_lsm_list_modules 461
+#define __NR_lsm_set_self_attr 460
 #define __NR_lstat 84
 #define __NR_lstat64 198
 #define __NR_madvise 119
@@ -339,6 +343,7 @@
 #define __NR_stat64 101
 #define __NR_statfs 99
 #define __NR_statfs64 298
+#define __NR_statmount 457
 #define __NR_statx 349
 #define __NR_stime 25
 #define __NR_swapoff 115
diff --git a/sysdeps/unix/sysv/linux/i386/arch-syscall.h b/sysdeps/unix/sysv/linux/i386/arch-syscall.h
index 21c1308bb3..cc775432d6 100644
--- a/sysdeps/unix/sysv/linux/i386/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/i386/arch-syscall.h
@@ -176,6 +176,7 @@
 #define __NR_link 9
 #define __NR_linkat 303
 #define __NR_listen 363
+#define __NR_listmount 458
 #define __NR_listxattr 232
 #define __NR_llistxattr 233
 #define __NR_lock 53
@@ -183,6 +184,9 @@
 #define __NR_lremovexattr 236
 #define __NR_lseek 19
 #define __NR_lsetxattr 227
+#define __NR_lsm_get_self_attr 459
+#define __NR_lsm_list_modules 461
+#define __NR_lsm_set_self_attr 460
 #define __NR_lstat 107
 #define __NR_lstat64 196
 #define __NR_madvise 219
@@ -386,6 +390,7 @@
 #define __NR_stat64 195
 #define __NR_statfs 99
 #define __NR_statfs64 268
+#define __NR_statmount 457
 #define __NR_statx 383
 #define __NR_stime 25
 #define __NR_stty 31
diff --git a/sysdeps/unix/sysv/linux/m68k/arch-syscall.h b/sysdeps/unix/sysv/linux/m68k/arch-syscall.h
index 6d788e3440..79f277dd5b 100644
--- a/sysdeps/unix/sysv/linux/m68k/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/m68k/arch-syscall.h
@@ -171,12 +171,16 @@
 #define __NR_link 9
 #define __NR_linkat 296
 #define __NR_listen 360
+#define __NR_listmount 458
 #define __NR_listxattr 229
 #define __NR_llistxattr 230
 #define __NR_lookup_dcookie 248
 #define __NR_lremovexattr 233
 #define __NR_lseek 19
 #define __NR_lsetxattr 224
+#define __NR_lsm_get_self_attr 459
+#define __NR_lsm_list_modules 461
+#define __NR_lsm_set_self_attr 460
 #define __NR_lstat 107
 #define __NR_lstat64 196
 #define __NR_madvise 238
@@ -373,6 +377,7 @@
 #define __NR_stat64 195
 #define __NR_statfs 99
 #define __NR_statfs64 263
+#define __NR_statmount 457
 #define __NR_statx 379
 #define __NR_stime 25
 #define __NR_swapoff 115
diff --git a/sysdeps/unix/sysv/linux/microblaze/arch-syscall.h b/sysdeps/unix/sysv/linux/microblaze/arch-syscall.h
index 91e1630f7b..779d5d5d70 100644
--- a/sysdeps/unix/sysv/linux/microblaze/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/microblaze/arch-syscall.h
@@ -176,6 +176,7 @@
 #define __NR_link 9
 #define __NR_linkat 303
 #define __NR_listen 348
+#define __NR_listmount 458
 #define __NR_listxattr 232
 #define __NR_llistxattr 233
 #define __NR_lock 53
@@ -183,6 +184,9 @@
 #define __NR_lremovexattr 236
 #define __NR_lseek 19
 #define __NR_lsetxattr 227
+#define __NR_lsm_get_self_attr 459
+#define __NR_lsm_list_modules 461
+#define __NR_lsm_set_self_attr 460
 #define __NR_lstat 107
 #define __NR_lstat64 196
 #define __NR_madvise 219
@@ -389,6 +393,7 @@
 #define __NR_stat64 195
 #define __NR_statfs 99
 #define __NR_statfs64 268
+#define __NR_statmount 457
 #define __NR_statx 398
 #define __NR_stime 25
 #define __NR_stty 31
diff --git a/sysdeps/unix/sysv/linux/mips/mips32/arch-syscall.h b/sysdeps/unix/sysv/linux/mips/mips32/arch-syscall.h
index d75af97467..86ffd5ce84 100644
--- a/sysdeps/unix/sysv/linux/mips/mips32/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/mips/mips32/arch-syscall.h
@@ -166,6 +166,7 @@
 #define __NR_link 4009
 #define __NR_linkat 4296
 #define __NR_listen 4174
+#define __NR_listmount 4458
 #define __NR_listxattr 4230
 #define __NR_llistxattr 4231
 #define __NR_lock 4053
@@ -173,6 +174,9 @@
 #define __NR_lremovexattr 4234
 #define __NR_lseek 4019
 #define __NR_lsetxattr 4225
+#define __NR_lsm_get_self_attr 4459
+#define __NR_lsm_list_modules 4461
+#define __NR_lsm_set_self_attr 4460
 #define __NR_lstat 4107
 #define __NR_lstat64 4214
 #define __NR_madvise 4218
@@ -362,6 +366,7 @@
 #define __NR_stat64 4213
 #define __NR_statfs 4099
 #define __NR_statfs64 4255
+#define __NR_statmount 4457
 #define __NR_statx 4366
 #define __NR_stime 4025
 #define __NR_stty 4031
diff --git a/sysdeps/unix/sysv/linux/mips/mips64/n32/arch-syscall.h b/sysdeps/unix/sysv/linux/mips/mips64/n32/arch-syscall.h
index 05bf7d251d..5d37a686e5 100644
--- a/sysdeps/unix/sysv/linux/mips/mips64/n32/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/mips/mips64/n32/arch-syscall.h
@@ -154,12 +154,16 @@
 #define __NR_link 6084
 #define __NR_linkat 6259
 #define __NR_listen 6049
+#define __NR_listmount 6458
 #define __NR_listxattr 6186
 #define __NR_llistxattr 6187
 #define __NR_lookup_dcookie 6206
 #define __NR_lremovexattr 6190
 #define __NR_lseek 6008
 #define __NR_lsetxattr 6181
+#define __NR_lsm_get_self_attr 6459
+#define __NR_lsm_list_modules 6461
+#define __NR_lsm_set_self_attr 6460
 #define __NR_lstat 6006
 #define __NR_madvise 6027
 #define __NR_map_shadow_stack 6453
@@ -332,6 +336,7 @@
 #define __NR_stat 6004
 #define __NR_statfs 6134
 #define __NR_statfs64 6217
+#define __NR_statmount 6457
 #define __NR_statx 6330
 #define __NR_swapoff 6163
 #define __NR_swapon 6162
diff --git a/sysdeps/unix/sysv/linux/mips/mips64/n64/arch-syscall.h b/sysdeps/unix/sysv/linux/mips/mips64/n64/arch-syscall.h
index 41ffaf3255..9b1e846e76 100644
--- a/sysdeps/unix/sysv/linux/mips/mips64/n64/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/mips/mips64/n64/arch-syscall.h
@@ -145,12 +145,16 @@
 #define __NR_link 5084
 #define __NR_linkat 5255
 #define __NR_listen 5049
+#define __NR_listmount 5458
 #define __NR_listxattr 5186
 #define __NR_llistxattr 5187
 #define __NR_lookup_dcookie 5206
 #define __NR_lremovexattr 5190
 #define __NR_lseek 5008
 #define __NR_lsetxattr 5181
+#define __NR_lsm_get_self_attr 5459
+#define __NR_lsm_list_modules 5461
+#define __NR_lsm_set_self_attr 5460
 #define __NR_lstat 5006
 #define __NR_madvise 5027
 #define __NR_map_shadow_stack 5453
@@ -313,6 +317,7 @@
 #define __NR_splice 5263
 #define __NR_stat 5004
 #define __NR_statfs 5134
+#define __NR_statmount 5457
 #define __NR_statx 5326
 #define __NR_swapoff 5163
 #define __NR_swapon 5162
diff --git a/sysdeps/unix/sysv/linux/nios2/arch-syscall.h b/sysdeps/unix/sysv/linux/nios2/arch-syscall.h
index d94e7e9ee9..abbc9ab6b0 100644
--- a/sysdeps/unix/sysv/linux/nios2/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/nios2/arch-syscall.h
@@ -130,12 +130,16 @@
 #define __NR_lgetxattr 9
 #define __NR_linkat 37
 #define __NR_listen 201
+#define __NR_listmount 458
 #define __NR_listxattr 11
 #define __NR_llistxattr 12
 #define __NR_llseek 62
 #define __NR_lookup_dcookie 18
 #define __NR_lremovexattr 15
 #define __NR_lsetxattr 6
+#define __NR_lsm_get_self_attr 459
+#define __NR_lsm_list_modules 461
+#define __NR_lsm_set_self_attr 460
 #define __NR_madvise 233
 #define __NR_map_shadow_stack 453
 #define __NR_mbind 235
@@ -290,6 +294,7 @@
 #define __NR_socketpair 199
 #define __NR_splice 76
 #define __NR_statfs64 43
+#define __NR_statmount 457
 #define __NR_statx 291
 #define __NR_swapoff 225
 #define __NR_swapon 224
diff --git a/sysdeps/unix/sysv/linux/powerpc/powerpc32/arch-syscall.h b/sysdeps/unix/sysv/linux/powerpc/powerpc32/arch-syscall.h
index b5522e8889..af0d2b121e 100644
--- a/sysdeps/unix/sysv/linux/powerpc/powerpc32/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/powerpc/powerpc32/arch-syscall.h
@@ -166,6 +166,7 @@
 #define __NR_link 9
 #define __NR_linkat 294
 #define __NR_listen 329
+#define __NR_listmount 458
 #define __NR_listxattr 215
 #define __NR_llistxattr 216
 #define __NR_lock 53
@@ -173,6 +174,9 @@
 #define __NR_lremovexattr 219
 #define __NR_lseek 19
 #define __NR_lsetxattr 210
+#define __NR_lsm_get_self_attr 459
+#define __NR_lsm_list_modules 461
+#define __NR_lsm_set_self_attr 460
 #define __NR_lstat 107
 #define __NR_lstat64 196
 #define __NR_madvise 205
@@ -374,6 +378,7 @@
 #define __NR_stat64 195
 #define __NR_statfs 99
 #define __NR_statfs64 252
+#define __NR_statmount 457
 #define __NR_statx 383
 #define __NR_stime 25
 #define __NR_stty 31
diff --git a/sysdeps/unix/sysv/linux/powerpc/powerpc64/arch-syscall.h b/sysdeps/unix/sysv/linux/powerpc/powerpc64/arch-syscall.h
index 162d782ae6..a4c70aa7fe 100644
--- a/sysdeps/unix/sysv/linux/powerpc/powerpc64/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/powerpc/powerpc64/arch-syscall.h
@@ -154,6 +154,7 @@
 #define __NR_link 9
 #define __NR_linkat 294
 #define __NR_listen 329
+#define __NR_listmount 458
 #define __NR_listxattr 215
 #define __NR_llistxattr 216
 #define __NR_lock 53
@@ -161,6 +162,9 @@
 #define __NR_lremovexattr 219
 #define __NR_lseek 19
 #define __NR_lsetxattr 210
+#define __NR_lsm_get_self_attr 459
+#define __NR_lsm_list_modules 461
+#define __NR_lsm_set_self_attr 460
 #define __NR_lstat 107
 #define __NR_madvise 205
 #define __NR_map_shadow_stack 453
@@ -352,6 +356,7 @@
 #define __NR_stat 106
 #define __NR_statfs 99
 #define __NR_statfs64 252
+#define __NR_statmount 457
 #define __NR_statx 383
 #define __NR_stime 25
 #define __NR_stty 31
diff --git a/sysdeps/unix/sysv/linux/riscv/rv32/arch-syscall.h b/sysdeps/unix/sysv/linux/riscv/rv32/arch-syscall.h
index 013222e5de..7315d164d6 100644
--- a/sysdeps/unix/sysv/linux/riscv/rv32/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/riscv/rv32/arch-syscall.h
@@ -117,12 +117,16 @@
 #define __NR_lgetxattr 9
 #define __NR_linkat 37
 #define __NR_listen 201
+#define __NR_listmount 458
 #define __NR_listxattr 11
 #define __NR_llistxattr 12
 #define __NR_llseek 62
 #define __NR_lookup_dcookie 18
 #define __NR_lremovexattr 15
 #define __NR_lsetxattr 6
+#define __NR_lsm_get_self_attr 459
+#define __NR_lsm_list_modules 461
+#define __NR_lsm_set_self_attr 460
 #define __NR_madvise 233
 #define __NR_map_shadow_stack 453
 #define __NR_mbind 235
@@ -268,6 +272,7 @@
 #define __NR_socketpair 199
 #define __NR_splice 76
 #define __NR_statfs64 43
+#define __NR_statmount 457
 #define __NR_statx 291
 #define __NR_swapoff 225
 #define __NR_swapon 224
diff --git a/sysdeps/unix/sysv/linux/riscv/rv64/arch-syscall.h b/sysdeps/unix/sysv/linux/riscv/rv64/arch-syscall.h
index d03dad8200..31a1130db9 100644
--- a/sysdeps/unix/sysv/linux/riscv/rv64/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/riscv/rv64/arch-syscall.h
@@ -122,12 +122,16 @@
 #define __NR_lgetxattr 9
 #define __NR_linkat 37
 #define __NR_listen 201
+#define __NR_listmount 458
 #define __NR_listxattr 11
 #define __NR_llistxattr 12
 #define __NR_lookup_dcookie 18
 #define __NR_lremovexattr 15
 #define __NR_lseek 62
 #define __NR_lsetxattr 6
+#define __NR_lsm_get_self_attr 459
+#define __NR_lsm_list_modules 461
+#define __NR_lsm_set_self_attr 460
 #define __NR_madvise 233
 #define __NR_map_shadow_stack 453
 #define __NR_mbind 235
@@ -277,6 +281,7 @@
 #define __NR_socketpair 199
 #define __NR_splice 76
 #define __NR_statfs 43
+#define __NR_statmount 457
 #define __NR_statx 291
 #define __NR_swapoff 225
 #define __NR_swapon 224
diff --git a/sysdeps/unix/sysv/linux/s390/s390-32/arch-syscall.h b/sysdeps/unix/sysv/linux/s390/s390-32/arch-syscall.h
index 98e6b68b31..cf8569304d 100644
--- a/sysdeps/unix/sysv/linux/s390/s390-32/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/s390/s390-32/arch-syscall.h
@@ -171,12 +171,16 @@
 #define __NR_link 9
 #define __NR_linkat 296
 #define __NR_listen 363
+#define __NR_listmount 458
 #define __NR_listxattr 230
 #define __NR_llistxattr 231
 #define __NR_lookup_dcookie 110
 #define __NR_lremovexattr 234
 #define __NR_lseek 19
 #define __NR_lsetxattr 225
+#define __NR_lsm_get_self_attr 459
+#define __NR_lsm_list_modules 461
+#define __NR_lsm_set_self_attr 460
 #define __NR_lstat 107
 #define __NR_lstat64 196
 #define __NR_madvise 219
@@ -372,6 +376,7 @@
 #define __NR_stat64 195
 #define __NR_statfs 99
 #define __NR_statfs64 265
+#define __NR_statmount 457
 #define __NR_statx 379
 #define __NR_stime 25
 #define __NR_swapoff 115
diff --git a/sysdeps/unix/sysv/linux/s390/s390-64/arch-syscall.h b/sysdeps/unix/sysv/linux/s390/s390-64/arch-syscall.h
index 951fbd7c97..f3536ed03f 100644
--- a/sysdeps/unix/sysv/linux/s390/s390-64/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/s390/s390-64/arch-syscall.h
@@ -146,12 +146,16 @@
 #define __NR_link 9
 #define __NR_linkat 296
 #define __NR_listen 363
+#define __NR_listmount 458
 #define __NR_listxattr 230
 #define __NR_llistxattr 231
 #define __NR_lookup_dcookie 110
 #define __NR_lremovexattr 234
 #define __NR_lseek 19
 #define __NR_lsetxattr 225
+#define __NR_lsm_get_self_attr 459
+#define __NR_lsm_list_modules 461
+#define __NR_lsm_set_self_attr 460
 #define __NR_lstat 107
 #define __NR_madvise 219
 #define __NR_map_shadow_stack 453
@@ -329,6 +333,7 @@
 #define __NR_stat 106
 #define __NR_statfs 99
 #define __NR_statfs64 265
+#define __NR_statmount 457
 #define __NR_statx 379
 #define __NR_swapoff 115
 #define __NR_swapon 87
diff --git a/sysdeps/unix/sysv/linux/sh/arch-syscall.h b/sysdeps/unix/sysv/linux/sh/arch-syscall.h
index 6b4418bcae..0c88bf10c7 100644
--- a/sysdeps/unix/sysv/linux/sh/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/sh/arch-syscall.h
@@ -164,12 +164,16 @@
 #define __NR_link 9
 #define __NR_linkat 303
 #define __NR_listen 343
+#define __NR_listmount 458
 #define __NR_listxattr 232
 #define __NR_llistxattr 233
 #define __NR_lookup_dcookie 253
 #define __NR_lremovexattr 236
 #define __NR_lseek 19
 #define __NR_lsetxattr 227
+#define __NR_lsm_get_self_attr 459
+#define __NR_lsm_list_modules 461
+#define __NR_lsm_set_self_attr 460
 #define __NR_lstat 107
 #define __NR_lstat64 196
 #define __NR_madvise 219
@@ -365,6 +369,7 @@
 #define __NR_stat64 195
 #define __NR_statfs 99
 #define __NR_statfs64 268
+#define __NR_statmount 457
 #define __NR_statx 383
 #define __NR_stime 25
 #define __NR_swapoff 115
diff --git a/sysdeps/unix/sysv/linux/sparc/sparc32/arch-syscall.h b/sysdeps/unix/sysv/linux/sparc/sparc32/arch-syscall.h
index 4f9460b1a3..19fa614624 100644
--- a/sysdeps/unix/sysv/linux/sparc/sparc32/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/sparc/sparc32/arch-syscall.h
@@ -169,12 +169,16 @@
 #define __NR_link 9
 #define __NR_linkat 292
 #define __NR_listen 354
+#define __NR_listmount 458
 #define __NR_listxattr 178
 #define __NR_llistxattr 179
 #define __NR_lookup_dcookie 208
 #define __NR_lremovexattr 182
 #define __NR_lseek 19
 #define __NR_lsetxattr 170
+#define __NR_lsm_get_self_attr 459
+#define __NR_lsm_list_modules 461
+#define __NR_lsm_set_self_attr 460
 #define __NR_lstat 40
 #define __NR_lstat64 132
 #define __NR_madvise 75
@@ -370,6 +374,7 @@
 #define __NR_stat64 139
 #define __NR_statfs 157
 #define __NR_statfs64 234
+#define __NR_statmount 457
 #define __NR_statx 360
 #define __NR_stime 233
 #define __NR_swapoff 213
diff --git a/sysdeps/unix/sysv/linux/sparc/sparc64/arch-syscall.h b/sysdeps/unix/sysv/linux/sparc/sparc64/arch-syscall.h
index 129ce50646..18516f20cb 100644
--- a/sysdeps/unix/sysv/linux/sparc/sparc64/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/sparc/sparc64/arch-syscall.h
@@ -152,12 +152,16 @@
 #define __NR_link 9
 #define __NR_linkat 292
 #define __NR_listen 354
+#define __NR_listmount 458
 #define __NR_listxattr 178
 #define __NR_llistxattr 179
 #define __NR_lookup_dcookie 208
 #define __NR_lremovexattr 182
 #define __NR_lseek 19
 #define __NR_lsetxattr 170
+#define __NR_lsm_get_self_attr 459
+#define __NR_lsm_list_modules 461
+#define __NR_lsm_set_self_attr 460
 #define __NR_lstat 40
 #define __NR_lstat64 132
 #define __NR_madvise 75
@@ -339,6 +343,7 @@
 #define __NR_stat64 139
 #define __NR_statfs 157
 #define __NR_statfs64 234
+#define __NR_statmount 457
 #define __NR_statx 360
 #define __NR_stime 233
 #define __NR_swapoff 213
diff --git a/sysdeps/unix/sysv/linux/syscall-names.list b/sysdeps/unix/sysv/linux/syscall-names.list
index aac065e7b3..6557bcfde4 100644
--- a/sysdeps/unix/sysv/linux/syscall-names.list
+++ b/sysdeps/unix/sysv/linux/syscall-names.list
@@ -21,8 +21,8 @@
 # This file can list all potential system calls.  The names are only
 # used if the installed kernel headers also provide them.
 
-# The list of system calls is current as of Linux 6.7.
-kernel 6.7
+# The list of system calls is current as of Linux 6.8.
+kernel 6.8
 
 FAST_atomic_update
 FAST_cmpxchg
@@ -239,6 +239,7 @@ lgetxattr
 link
 linkat
 listen
+listmount
 listxattr
 llistxattr
 llseek
@@ -247,6 +248,9 @@ lookup_dcookie
 lremovexattr
 lseek
 lsetxattr
+lsm_get_self_attr
+lsm_list_modules
+lsm_set_self_attr
 lstat
 lstat64
 madvise
@@ -593,6 +597,7 @@ stat
 stat64
 statfs
 statfs64
+statmount
 statx
 stime
 stty
diff --git a/sysdeps/unix/sysv/linux/x86_64/64/arch-syscall.h b/sysdeps/unix/sysv/linux/x86_64/64/arch-syscall.h
index 4fa5b942c5..b122216013 100644
--- a/sysdeps/unix/sysv/linux/x86_64/64/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/x86_64/64/arch-syscall.h
@@ -149,12 +149,16 @@
 #define __NR_link 86
 #define __NR_linkat 265
 #define __NR_listen 50
+#define __NR_listmount 458
 #define __NR_listxattr 194
 #define __NR_llistxattr 195
 #define __NR_lookup_dcookie 212
 #define __NR_lremovexattr 198
 #define __NR_lseek 8
 #define __NR_lsetxattr 189
+#define __NR_lsm_get_self_attr 459
+#define __NR_lsm_list_modules 461
+#define __NR_lsm_set_self_attr 460
 #define __NR_lstat 6
 #define __NR_madvise 28
 #define __NR_map_shadow_stack 453
@@ -321,6 +325,7 @@
 #define __NR_splice 275
 #define __NR_stat 4
 #define __NR_statfs 137
+#define __NR_statmount 457
 #define __NR_statx 332
 #define __NR_swapoff 168
 #define __NR_swapon 167
diff --git a/sysdeps/unix/sysv/linux/x86_64/x32/arch-syscall.h b/sysdeps/unix/sysv/linux/x86_64/x32/arch-syscall.h
index b9db8bc5be..3040a47d72 100644
--- a/sysdeps/unix/sysv/linux/x86_64/x32/arch-syscall.h
+++ b/sysdeps/unix/sysv/linux/x86_64/x32/arch-syscall.h
@@ -143,12 +143,16 @@
 #define __NR_link 1073741910
 #define __NR_linkat 1073742089
 #define __NR_listen 1073741874
+#define __NR_listmount 1073742282
 #define __NR_listxattr 1073742018
 #define __NR_llistxattr 1073742019
 #define __NR_lookup_dcookie 1073742036
 #define __NR_lremovexattr 1073742022
 #define __NR_lseek 1073741832
 #define __NR_lsetxattr 1073742013
+#define __NR_lsm_get_self_attr 1073742283
+#define __NR_lsm_list_modules 1073742285
+#define __NR_lsm_set_self_attr 1073742284
 #define __NR_lstat 1073741830
 #define __NR_madvise 1073741852
 #define __NR_mbind 1073742061
@@ -312,6 +316,7 @@
 #define __NR_splice 1073742099
 #define __NR_stat 1073741828
 #define __NR_statfs 1073741961
+#define __NR_statmount 1073742281
 #define __NR_statx 1073742156
 #define __NR_swapoff 1073741992
 #define __NR_swapon 1073741991
